### PR TITLE
Off-hand/shield slot functional, save and load slot, bow + arrow functional (fixes #3714)

### LINKED ISSUE #3714: Puting a item in shield slot shows "Slot number higher than available window slots..."
Client version:   1.11.2 / 1.10
Server OS:        Debian (ARM)
Commit id:       commit id: approx: 7c4576a built at: approx: 2017-05-09 14:24:41 +0200

### Expected behavior
You should be able to put items at shield slot without warns in console. Bow + arrow should work.

### Actual behavior
Every click on shield slot Cuberite shows in console:
"Slot number higher than available window slots: 45, max 45 received from "<player_name>"; ignoring."
In addition when you try to throw arrows with a bow doesn't work. The arrow count decreases but no arrow is shot. Arrows dissapear but they appear again on slot 1 when open inventory and click on slot where there they were.

### Steps to reproduce the behavior
Just click on shield slot and loot at the console.
Equip yourself with a bow and put arrows in slot 1 and try to show arrows.

### PR DESCRIPTION
**Pull request by spekdrum**
This adds these functionalities:
- You can put items in off-hand/shield slot so they don't fall to floor.
- Save / load of inventory slot off-hand/shield.
- Arrow shooting with bow in off-hand slot.

I did several tests. Putting arrows in off-hand slot and hotbar, putting bow in off-hand and arrows in hand, hotbar, and it works. I also tryed with 1.8 protocol, where there is no off-hand/shield slot, no crashes so far.

I didn't test shields, maybe tomorrow!

### CODE REVIEW

File: src/ClientHandle.cpp [hunk 1]
```diff
@@ -1524,6 +1529,11 @@ void cClientHandle::HandleRightClick(int a_BlockX, int a_BlockY, int a_BlockZ, e
 		ItemHandler->OnItemUse(World, m_Player, PluginInterface, Equipped, a_BlockX, a_BlockY, a_BlockZ, a_BlockFace);
 		PlgMgr->CallHookPlayerUsedItem(*m_Player, a_BlockX, a_BlockY, a_BlockZ, a_BlockFace, a_CursorX, a_CursorY, a_CursorZ);
 	}
+	// Charge bow when it's in slot off-hand / shield
+	if ((a_BlockFace == BLOCK_FACE_NONE && m_Player->GetInventory().GetShieldSlot().m_ItemType == E_ITEM_BOW))
```
Comment (Pokechu22) [inline src/ClientHandle.cpp]: This should be

```c++
if ((a_BlockFace == BLOCK_FACE_NONE) && (m_Player->GetInventory().GetShieldSlot().m_ItemType == E_ITEM_BOW))
```

Comment (spekdrum) [inline src/ClientHandle.cpp]: Thanks, I'll continue with this tonight, time to work :\

**Comment by spekdrum** [comment]
Sorry but I don't understand this travis-ci issues. I documented them, but keeps spoting errors. Can anyone help me? I'm not very familiar to C++ standards (I'm a Java coder). Thanks!
```
ERROR: Newly undocumented API symbols found:
cInventory.GetShieldSlot (function)
cInventory.SetShieldSlot (function)
cInventory.invShieldCount (constant)
cInventory.invShieldOffset (constant)
```

**Comment by Seadragon91** [comment]
The documentation for `cInventory` is in file [APIDesc.lua](https://github.com/cuberite/cuberite/blob/master/Server/Plugins/APIDump/APIDesc.lua). Just look at the other functions and then it should be easy to add them.

**Comment by Seadragon91** [comment]
The documentation is created manually. It's used to create the api-doc website and I use it for example fro a fuzzing plugin.

**Comment by spekdrum** [comment]
Thanks, I didn't know about that file (lol)

File: Server/Plugins/APIDump/APIDesc.lua [hunk 2]
```diff
@@ -7970,6 +7970,16 @@ These ItemGrids are available in the API and can be manipulated by the plugins,
...
 				},
+				GetShieldSlot =
+				{
+					Returns =
+					{
+						{
+							Type = "cItem",
+						},
+					},
+					Notes = "Returns current item in shield slot. Note that the returned item is read-only",
```
Comment (Seadragon91) [inline Server/Plugins/APIDump/APIDesc.lua]: Remove the read-only addition,.

File: Server/Plugins/APIDump/APIDesc.lua [hunk 3]
```diff
@@ -8177,6 +8187,17 @@ These ItemGrids are available in the API and can be manipulated by the plugins,
 					},
 					Notes = "Sets the specified hotbar slot contents",
 				},
+				SetShieldSlot =
+				{
+					Params =
+					{
+						{
+							Name = "cItem",
```
Comment (Seadragon91) [inline Server/Plugins/APIDump/APIDesc.lua]: Item

File: Server/Plugins/APIDump/APIDesc.lua [hunk 4]
```diff
@@ -8221,7 +8242,7 @@ These ItemGrids are available in the API and can be manipulated by the plugins,
 				invHotbarCount =
 				{
 					Notes = "Number of slots in the Hotbar part",
-				},
+				},
```
Comment (Seadragon91) [inline Server/Plugins/APIDump/APIDesc.lua]: Remove spaces.

**Comment by Seadragon91** [comment]
Confirmed that placing shield, bow works and shooting with the bow.

**Comment by madmaxoft** [comment]
Nice going. Thanks for the contributions :)

**Comment by spekdrum** [comment]
U r welcome, so glad to help :D

**Pull request merged**